File: android/app/src/main/kotlin/com/dicyvpn/android/StatusSink.kt
package com.dicyvpn.android

import io.flutter.plugin.common.EventChannel.EventSink

/**
 * Wrapper for a generic EventSink for type safety
 */
class StatusSink(private val sink: EventSink) {
    fun success(event: Status) {
        sink.success(event.name)
    }

    fun error(errorCode: String?, errorMessage: String?, errorDetails: Any?) {
        sink.error(errorCode, errorMessage, errorDetails)
    }

    fun endOfStream() {
        sink.endOfStream()
    }
}

File: ios/Runner/StatusSink.swift
//
//  StatusSink.swift
//  Runner
//
//  Created by user258751 on 5/17/24.
//

import Flutter

class StatusSink {
    private var sink: FlutterEventSink
    
    init(sink: @escaping FlutterEventSink) {
        self.sink = sink
    }
    
    func success(event: Status) {
        sink(event.rawValue)
    }
    
    func error(errorCode: String, errorMessage: String?, errorDetails: Any?) {
        sink(FlutterError(code: errorCode, message: errorMessage, details: errorDetails))
    }
    
    func endOfStream() {
        sink(FlutterEndOfEventStream)
    }
}
